app/Models/SnipeSCIMConfig.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Helper;
+use ArieTimmerman\Laravel\SCIMServer\SCIM\Schema;
+use ArieTimmerman\Laravel\SCIMServer\Attribute\AttributeMapping;
+
+
+class SnipeSCIMConfig extends \ArieTimmerman\Laravel\SCIMServer\SCIMConfig
+{
+    public function __constructor(...$params)
+    {
+        \Log::info("CONSTRUCTOR CALLED!!@!!!! FOr my config thing");
+        self::super(...$params);
+    }
+
+    public function getUserConfig()
+    {
+        // Much of this is copied verbatim from the library, then adjusted for our needs
+        return [
+                
+            // Set to 'null' to make use of auth.providers.users.model (App\User::class)
+            'class' => User::class,
+            
+            'validations' => [
+    
+                'urn:ietf:params:scim:schemas:core:2.0:User:userName' => 'required',
+                'urn:ietf:params:scim:schemas:core:2.0:User:password' => 'nullable',
```

Comment:
Our passwords are not nullable - will this conflict? 

This brings up a different question - will this somehow be able to bypass the model level validation and/or settings validation? If it can, we’re in a world of hurt, since we’ll enable adding bad data, which can mean saving an object will fail later in a way we don’t expect.

Reply:
*GREAT* question. I don’t know. I don’t have a good SCIM test-rig, but if I can figure one out, I’ll test this - and confirm or reject it.

Model-level validation **should** fail if given a model that is invalid, so I’d _hope_ that’d be caught. But hope is not certainty.

Reply:
Right, but how we fail is important here. If we don’t have a way to communicate the errors, it will add to our support load (and confusion).

Reply:
For the major SCIM providers I know of, SCIM will never provide a password to the application. Generally, the passwords wont be stored with reversible encryption, and will not be provided to an SCIM endpoint.

The password would need to be nullable in the model/database. The user would only be able to successfully login with SSO at that point.

Alternatively, a welcome email could be sent instructing the user to login with a randomly generated password to access Snipe IT, or provide an activation link to enter a new password. That's how I have seen other applications implement this without SSO.

Reply:
Fair enough. If we need to, we can hardcode a 'xxxxxxx' password which is not possible to hash towards. We'll have to see how it stands up when Azure tries to SCIM us up, I guess?